src/metabase/api/user.clj

```diff
@@ -78,7 +78,8 @@
   {password     [Required ComplexPassword]}
   (check-self-or-superuser id)
   (let-404 [user (db/select-one [User :password_salt :password], :id id, :is_active true)]
-    (when (= id *current-user-id*)
+    (when (and (not (:is_superuser @*current-user*))
+               (= id *current-user-id*))
```

Comment:
In this case it does.  The api call as a whole is only meant to be available to any superuser or the api calling user regardless of their superuser status.  So that check continues to make sense.

The additional line a bit lower down is applying an extra check only in the case that the api call was  made by a non-superuser.

So both checks make sense.